main.tf

```diff
@@ -158,6 +158,11 @@ resource "azurerm_virtual_machine" "instance" {
 }
 
 resource "null_resource" "instance-prereq" {
+  # Changes to any instance of the cluster requires re-provisioning
+  triggers {
+    current_ec2_instance_id = "${element(azurerm_virtual_machine.instance.*.id, count.index)}"
```

Comment:
not ec2 instance. Should be renamed

Reply:
Updated the variable name and made it simpler to keep it consistent with all the cloud providers.